src/curry/memo.ts

```diff
@@ -54,9 +27,33 @@ export interface MemoOptions<TArgs extends any[]> {
  * // calls === [10, 11]
  * ```
  */
-export function memo<TArgs extends any[], TResult>(
-  func: (...args: TArgs) => TResult,
-  options: MemoOptions<TArgs> = {},
-): (...args: TArgs) => TResult {
-  return memoize({}, func, options.key ?? null, options.ttl ?? null)
+export function memo<Fn extends (...args: any[]) => any>(
+  func: Fn,
+  options: MemoOptions<Fn> = {},
+): Memoized<Fn> {
+  const { key: keyFunc, ttl, cache = {} } = options
+
+  return function callWithMemo(...args) {
+    const key = keyFunc ? keyFunc(...args) : JSON.stringify({ args })
```

Comment:
`JSON.stringify` doesn't preserve references, which doesn't look like expected default behavior. I expect next code to be call `fn` only once:

```js
const obj = {};

const memoizedFn = memo(fn);

memoizedFn(obj);
memoizedFn(obj);
```

At least, this is how it works in different popular libraries like memoize. But memoize uses only a first arg as a key. 

We can implement solution based on the trie data-structure (like https://github.com/Minhir/multikeys) with radashi traverse function help.

Reply:
I didn't change this part at all. This is how Radash `memo` works. I'm open to a PR that makes this part more robust though, if you're willing!